src/Network/Transport/TCP.hs

```diff
@@ -1450,7 +1452,7 @@ schedule theirEndPoint act = do
 -- 'runScheduledAction' or by another).
 runScheduledAction :: EndPointPair -> Action a -> IO a
 runScheduledAction (ourEndPoint, theirEndPoint) mvar = do
-    join $ readChan (remoteScheduled theirEndPoint)
+    join $ uninterruptibleMask_ $ readChan (remoteScheduled theirEndPoint)
```

Comment:
> I'd prefer setting exception handlers or using resourcet rather than masking whole sections of code. 

This code is a bit trickier than the code that `resourcet` or `bracket` solve (here we from 1 to 2 threads where we are allocating resource (`schedule`), and at the end we need to execute an action in order, but not exactly the action that we have `scheduled`, otherwise we will have to introduce lock that will lead to a distributed deadlock, that was solved by `schedule`/`runScheduledAction` pair).
I think it's possible to use `resourcet` here, but it will call `runScheduledAction` as a finalizer and will have exactly the same problem that we have discussed here. And code will be much trickier than we have now.